woocommerce/woocommerce-ios
# [Mobile Payments] Badge tap to pay on eligible devices and stores, until Set up Tap to Pay opened

Closes: #9582


## Description

N.B. this follows #9801, but changes the logic and display details substantially. If you'd prefer to review both together, feel free to change the base of this PR to `trunk`.

We've added badges leading to Set up Tap to Pay on iPhone, for eligible devices and stores. This PR resolves the three outstanding logic issues from #9801:

- The badges will be shown on any store or device, regardless of Tap to Pay support or previous activity.
- After removal, the badges will return the next time you open the app.
- Taking a Tap to Pay payment will not remove the badges

Now the badges will only be shown when you are using a supported device, an eligible store, have not previously used Tap to Pay to take a transaction, and will remember when the badges have been dismissed.

## Testing instructions


Using an iPhone XS or newer on iOS 16 or above, and a US-based store with WCPay or the Stripe extension

1. Delete then reinstall the app
2. Open the app
3. Observe that the badges are present on the `Menu` tab, and the `Payments` row in the menu, and the `Set up Tap to Pay on iPhone` row in the IPP menu
4. Tap `Set up Tap to Pay on iPhone`, then tap back
5. Observe that all the badges are gone
6. Force quit then re-open the app
7. Observe the badges are still gone


Using an iPhone XS or newer on iOS 16 or above, and a US-based store with WCPay or the Stripe extension

1. Delete then reinstall the app
2. Open the app
3. Observe that the badges are present on the `Menu` tab, and the `Payments` row in the menu, and the `Set up Tap to Pay on iPhone` row in the IPP menu
4. Tap `Collect Payment` and collect a payment using Tap to Pay on iPhone
5. Observe that all the badges are gone

Using an iPhone X or older, or an iPad, or any device running iOS 15, and a US-based store with WCPay or the Stripe extension

1. Delete then reinstall the app
2. Open the app
3. Observe that the badges are not present on the `Menu` tab, and the `Payments` row in the menu, and the `Set up Tap to Pay on iPhone` row in the IPP menu

Using an iPhone XS or newer on iOS 16 or above, and a CA or GB-based store with WCPay

1. Delete then reinstall the app
2. Open the app
3. Observe that the badges are not present on the `Menu` tab, and the `Payments` row in the menu, and the `Set up Tap to Pay on iPhone` row in the IPP menu




## Screenshots



---
